### PR TITLE
feat: add personal builder page

### PR DESCRIPTION
**Pull request by gboigwe**
## Description

Added some information about me

## Additional Information

- [x] I have read the [contributing docs](/scaffold-eth/scaffold-eth-2/blob/main/CONTRIBUTING.md) (if this is your first contribution)
- [x] This is not a duplicate of any [existing pull request](https://github.com/scaffold-eth/scaffold-eth-2/pulls)

## Related Issues

_Closes #{2}_

Your ENS/address: 0x5cc8Be96B1C9A68F57a73b5bEa60cF5D890055A1

### CODE REVIEW

File: packages/nextjs/app/builders/0x5cc8Be96B1C9A68F57a73b5bEa60cF5D890055A1/page.tsx [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+  name: "text-3xl font-bold",
+  address: "font-mono text-sm text-base-content/70",
+  section: "mt-8",
+  sectionTitle: "text-xl font-semibold mb-4 text-base-content",
+  bio: "text-base-content/80 leading-relaxed",
+  socialLinks: "flex gap-4 items-center justify-center md:justify-start",
+  socialIcon: "p-2 hover:bg-base-200 rounded-full transition-colors duration-200 w-10 h-10",
+};
+
+const BuilderPage = () => {
```
Comment (phipsae) [inline packages/nextjs/app/builders/0x5cc8Be96B1C9A68F57a73b5bEa60cF5D890055A1/page.tsx:19]: You might want to give your page a more descriptive name (we will have more builder pages) and consider typing your page as NextPage since we are using TypeScript, as this ensures better integration and adds clarity for future maintainers by indicating that the component is a Next.js page.

File: packages/nextjs/app/builders/0x5cc8Be96B1C9A68F57a73b5bEa60cF5D890055A1/page.tsx [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+  const builderDetails = {
```
Comment (phipsae) [inline packages/nextjs/app/builders/0x5cc8Be96B1C9A68F57a73b5bEa60cF5D890055A1/page.tsx:20]: You could place the const outside of the component since it is a static profile page.

File: packages/nextjs/app/builders/0x5cc8Be96B1C9A68F57a73b5bEa60cF5D890055A1/page.tsx [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+    name: "Gbolahan Akande",
+    address: "0x5cc8Be96B1C9A68F57a73b5bEa60cF5D890055A1",
+    bio: "A passionate Web3 developer with experience in Solidity and full-stack development. Currently building decentralized applications and contributing to BuidlGuidl projects. Excited about the future of blockchain technology and its potential to reshape the digital world.",
```
Comment (phipsae) [inline packages/nextjs/app/builders/0x5cc8Be96B1C9A68F57a73b5bEa60cF5D890055A1/page.tsx:23]: Some highlighting in the text might be nice.

File: packages/nextjs/app/builders/0x5cc8Be96B1C9A68F57a73b5bEa60cF5D890055A1/page.tsx [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+  };
+
+  return (
+    <div className={styles.container}>
+      <div className={styles.card}>
+        <div className={styles.profileSection}>
+          <img src={builderDetails.avatar} alt="Profile" className={styles.avatar} />
+          <div className={styles.nameSection}>
+            <h1 className={styles.name}>{builderDetails.name}</h1>
+            <p className={styles.address}>{builderDetails.address}</p>
```
Comment (phipsae) [inline packages/nextjs/app/builders/0x5cc8Be96B1C9A68F57a73b5bEa60cF5D890055A1/page.tsx:39]: You could consider using the SE2 component `<Address>`

**Review by phipsae — CHANGES_REQUESTED** [review]
Thanks a lot for the page! 

Below you'll find some little comments.

**Comment by gboigwe** [comment]
# Personal Builder Page

## Description
Added my personal builder page with the following features:
- Responsive design for mobile and desktop
- Dark/light mode support
- Enhanced typography and visual hierarchy
- Social links integration
- Address display using SE2 components

## Screenshots
image

## Additional Notes
- Implemented all feedback from previous review
- Used existing Scaffold-ETH 2 components
- Maintained consistent styling with the project

**Comment by gboigwe** [comment]
# Enhanced Homepage and Builders Grid

## Changes Made
- ✨ Improved homepage dark mode visibility and contrast
- 🏗️ Added BuildersGrid component showing registered builders
- 🔄 Created API route to fetch builder addresses from filesystem
- 🔗 Connected grid to personal builder pages (/builders/0x...)
- 💅 Updated styling and layout for better UX

## Implementation Details
- Homepage now reads checkedInCounter from BatchRegistry contract
- BuildersGrid scans filesystem for builder pages in `/builders` directory
- Added loading states and error handling
- Maintained consistent styling with existing components

## Screenshots
### The checked in count:
image

### The builders profile grid:
image
- For now I only have mine, but if this PR is accepted and merged, then every builder would be registered in.

## Testing Done
- Tested dark mode visibility
- Verified builder count display
- Confirmed grid updates when new builder pages are added
- Checked mobile responsiveness

## Additional Notes
- Grid automatically updates as builders add their profile pages
- Uses existing Address component for consistency

File: packages/nextjs/app/page.tsx
Comment (phipsae) [inline packages/nextjs/app/page.tsx:1]: I know in the beginning it is a little bit confusing. For each and every issue you tackle there should be a separate PR and branch. So it's easier for reviewers to understand what's going on. 

And this isn't related to your personal page PR, so please leave that out.

File: packages/nextjs/components/batch/BuildersGrid.tsx
Comment (phipsae) [inline packages/nextjs/components/batch/BuildersGrid.tsx:1]: I know in the beginning it is a little bit confusing. For each and every issue you tackle there should be a separate PR and branch. So it's easier for reviewers to understand what's going on. 

And this isn't related to your personal page PR, so please leave that out.

File: packages/nextjs/app/api/builders/route.ts
Comment (phipsae) [inline packages/nextjs/app/api/builders/route.ts:1]: I know in the beginning it is a little bit confusing. For each and every issue you tackle there should be a separate PR and branch. So it's easier for reviewers to understand what's going on. 

And this isn't related to your personal page PR, so please leave that out.

**Review by phipsae — CHANGES_REQUESTED** [review]
Thanks for the changes and also the implementation of builders overview!
 
Unfortunately that has nothing to do with your personal page PR, pls take that out and create a dedicated PR for that. 

Thanks a lot!

File: packages/nextjs/app/page.tsx
Comment (gboigwe) [reply packages/nextjs/app/page.tsx:1]: I have just reset and forced push up to only my personal page

File: packages/nextjs/components/batch/BuildersGrid.tsx
Comment (gboigwe) [reply packages/nextjs/components/batch/BuildersGrid.tsx:1]: I have just reset and forced pushed up to only my personal page

File: packages/nextjs/app/api/builders/route.ts
Comment (gboigwe) [reply packages/nextjs/app/api/builders/route.ts:1]: I have just reset and forced pushed up to only my personal page

**Comment by gboigwe** [comment]
Please confirm now that the PR has only solved the issue pertaining to it. Thank you

**Comment by phipsae** [comment]
Perfect, no its good, thanks! Lets merge!

**Comment by gboigwe** [comment]
> Perfect, no its good, thanks! Lets merge! 

Thank you, It's a real great relief comment. Thanks.
